braindecode/datasets/chb_mit.py

```diff
@@ -0,0 +1,206 @@
+         last ``_`` before the extension. E.g., ``'channels'``.
+         The following filename suffix's are accepted:
+         'meg', 'markers', 'eeg', 'ieeg', 'T1w',
+         'participants', 'scans', 'electrodes', 'coordsystem',
+         'channels', 'events', 'headshape', 'digitizer',
+         'beh', 'physio', 'stim'
+     extensions : str | array-like of str | None
+         The extension of the filename. E.g., ``'.json'``.
+         By default, uses the ones accepted by :func:`mne_bids.read_raw_bids`.
+     datatypes : str | array-like of str | None
+         The BIDS data type, e.g., ``'anat'``, ``'func'``, ``'eeg'``, ``'meg'``,
+         ``'ieeg'``.
+     check : bool
+         If ``True``, only returns paths that conform to BIDS. If ``False``
+         (default), the ``.check`` attribute of the returned
+         :class:`mne_bids.BIDSPath` object will be set to ``True`` for paths that
+         do conform to BIDS, and to ``False`` for those that don't.
+     preload : bool
+         If True, preload the data. Defaults to False.
+     n_jobs : int
+         Number of jobs to run in parallel. Defaults to 1.
+
+     References
+     ----------
+     .. [Guttag2010] Guttag, J. (2010). CHB-MIT Scalp EEG Database (version 1.0.0).
+         PhysioNet. RRID:SCR_007345. https://doi.org/10.13026/C2K01R
+     .. [Shoeb2009] Ali Shoeb. Application of Machine Learning to Epileptic
+         Seizure Onset Detection and Treatment. PhD Thesis,
+         Massachusetts Institute of Technology, September 2009.
+     .. [Dan2025] Dan, J., Pale, U., Amirshahi, A., Cappelletti, W.,
+         Ingolfsson, T. M., Wang, X., ... & Ryvlin, P. (2025).
+         SzCORE: seizure community open-source research evaluatio
+         framework for the validation of electroencephalography-based
+         automated seizure detection algorithms. Epilepsia, 66, 14-24.
+    """
+
+    def __init__(self, root=None, *args, **kwargs):
+        # correct the path if needed
+        path_root = root
+        if path_root is not None:
+            list_tsv = glob.glob(f"{path_root}/**/participants.tsv", recursive=True)
+            if isinstance(list_tsv, list) and len(list_tsv) > 0:
+                path_root = Path(list_tsv[0]).parent
+
+        # Download dataset if not present
+        if path_root is None or len(list_tsv) == 0:
+            path_root = fetch_dataset(
+                dataset_params=CHB_MIT_dataset_params,
+                path=Path(path_root) if path_root is not None else None,
+                processor="unzip",
+                force_update=False,
+            )
+            # First time we fetch the dataset, we need to move the files to the
+            # correct directory.
+            path_root = _correct_path(path_root)
+
+        kwargs["root"] = path_root
+
+        super().__init__(
+            *args,
+            extensions=".edf",
+            check=False,
+            **kwargs,
+        )
+
+
+def _correct_path(path: str):
```

Comment:
This could be a staticmethod if the dataset if it only applies to this one. WDYT?